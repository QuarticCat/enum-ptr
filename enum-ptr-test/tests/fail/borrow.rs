#![allow(dead_code)]

use enum_ptr::{EnumPtr, Unit};

#[derive(EnumPtr)]
#[enum_ptr(borrow)]
#[repr(C, usize)]
enum Foo<'a> {
    A(&'a i32),
    B(Unit),
}

#[derive(EnumPtr)]
#[enum_ptr(borrow)]
#[repr(C, usize)]
enum Bar<'a> {
    A(&'a i32),
    #[enum_ptr(skip)]
    B(Unit),
}

#[derive(EnumPtr)]
#[enum_ptr(borrow(rename = "RefBaz"))]
#[repr(C, usize)]
enum Baz<'a> {
    A(&'a i32),
    #[enum_ptr(skip_borrow)]
    B(Unit),
}

fn main() {}
