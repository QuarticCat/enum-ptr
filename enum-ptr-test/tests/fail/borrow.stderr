error[E0277]: the trait bound `Unit: FieldDeref` is not satisfied
 --> tests/fail/borrow.rs:5:10
  |
5 | #[derive(EnumPtr)]
  |          ^^^^^^^ the trait `FieldDeref` is not implemented for `Unit`
  |
  = help: the following other types implement trait `FieldDeref`:
            &T
            &mut T
            Arc<T>
            Box<T>
            Option<&T>
            Option<&mut T>
            Option<Arc<T>>
            Option<Box<T>>
          and $N others
  = note: this error originates in the derive macro `EnumPtr` (in Nightly builds, run with -Z macro-backtrace for more info)
